ambry-server/src/main/java/com/github/ambry/server/mysql/AccountStatsMySqlStore.java

```diff
@@ -0,0 +1,190 @@
+    accountReportsDao = new AccountReportsDao(dataAccessor, clustername, hostname);
+    storeMetrics = new AccountStatsMySqlStore.Metrics(registry);
+    if (!Strings.isNullOrEmpty(localBackupFilePath)) {
+      // load backup file and this backup is the previous stats
+      ObjectMapper objectMapper = new ObjectMapper();
+      try {
+        this.previousStats = objectMapper.readValue(new File(localBackupFilePath), StatsWrapper.class);
+      } catch (Exception e) {
+        this.previousStats = null;
+      }
+    }
+  }
+
+  /**
+   * Publish the {@link StatsWrapper} to mysql database. This method ignores the error information from {@link StatsWrapper}
+   * and only publish the container storage usages that are different from the previous one.
+   * @param statsWrapper The {@link StatsWrapper} to publish.
+   */
+  public void publish(StatsWrapper statsWrapper) {
+    if (previousStats == null) {
+      applyFunctionToContainerUsageInStatsSnapshot(statsWrapper.getSnapshot(), accountReportsDao::updateStorageUsage);
+    } else {
+      applyFunctionToContainerUsageInDifferentStatsSnapshots(statsWrapper.getSnapshot(), previousStats.getSnapshot(),
+          accountReportsDao::updateStorageUsage);
+    }
+    previousStats = statsWrapper;
+  }
+
+  StatsWrapper getPreviousStats() {
+    return previousStats;
+  }
+
+  private void applyFunctionToContainerUsageInStatsSnapshot(StatsSnapshot statsSnapshot, ContainerUsageFunction func) {
+    int batchSize = 0;
+    long startTimeMs = System.currentTimeMillis();
+    // StatsSnapshot has three levels, Parition -> Account -> Container
+    Map<String, StatsSnapshot> partitionMap = statsSnapshot.getSubMap();
+    for (Map.Entry<String, StatsSnapshot> partitionMapEntry : partitionMap.entrySet()) {
+      String partitionIdKey = partitionMapEntry.getKey();
+      StatsSnapshot accountStatsSnapshot = partitionMapEntry.getValue();
+      short partitionId = Short.valueOf(partitionIdKey.substring("Partition[".length(), partitionIdKey.length() - 1));
+      Map<String, StatsSnapshot> accountMap = accountStatsSnapshot.getSubMap();
+      for (Map.Entry<String, StatsSnapshot> accountMapEntry : accountMap.entrySet()) {
+        String accountIdKey = accountMapEntry.getKey();
+        StatsSnapshot containerStatsSnapshot = accountMapEntry.getValue();
+        short accountId = Short.valueOf(accountIdKey.substring(2, accountIdKey.length() - 1));
+        Map<String, StatsSnapshot> containerMap = containerStatsSnapshot.getSubMap();
+        for (Map.Entry<String, StatsSnapshot> containerMapEntry : containerMap.entrySet()) {
+          String containerIdKey = containerMapEntry.getKey();
+          short containerId = Short.valueOf(containerIdKey.substring(2, containerIdKey.length() - 1));
+          long storageUsage = containerMapEntry.getValue().getValue();
+          func.apply(partitionId, accountId, containerId, storageUsage);
+          batchSize++;
```

Comment:
I am seeing `addBatch` method from `java.sql.Statement`. There might be a way to batch those writes to mysql database later. Besides, I also want to know how many container has changed after last update. This would give us a way to know the write traffic to mysql database.